src/client.rs

```diff
@@ -95,3 +90,95 @@ impl SpiceClient {
         self.firecache.query(query).await
     }
 }
+
+/// Builder for creating a `SpiceClient`.
+///
+/// By default the `SpiceClient` will use local spice runtime flight endpoint.
+/// Follow [spiceai quickstart](https://github.com/spiceai/spiceai?tab=readme-ov-file#%EF%B8%8F-quickstart-local-machine) to setup local spice runtime.
+/// ```
+/// # use spiceai::ClientBuilder;
+/// #
+/// # #[tokio::main]
+/// # async fn main() {
+/// #    let mut client = ClientBuilder::new().build();
+/// # }
+/// ```
+/// To use default Spice.ai Cloud endpoints, you can use the `with_spiceai_cloud()` method.
+///
+/// ```
+/// # use spiceai::ClientBuilder;
+/// #
+/// # #[tokio::main]
+/// # async fn main() {
+/// #    let mut client = ClientBuilder::new()
+/// #      .with_api_key("API_KEY")
+/// #      .with_spiceai_cloud()
+/// #      .build();
+/// # }
+/// ```
+///
+pub struct SpiceClientBuilder {
+    api_key: Option<String>,
+    firecache_url: Option<String>,
+    flight_url: Option<String>,
+}
+
+impl Default for SpiceClientBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl SpiceClientBuilder {
+    pub fn new() -> Self {
+        Self {
+            api_key: None,
+            firecache_url: None,
+            flight_url: None,
+        }
+    }
+
+    /// Configures the `SpiceClient` to use the given API key.
+    pub fn with_api_key(mut self, api_key: &str) -> Self {
```

Comment:
```suggestion
    pub fn api_key(mut self, api_key: &str) -> Self {
```

The [builder pattern](https://rust-unofficial.github.io/patterns/patterns/creational/builder.html) doesn't use `with_` prefixes for setting options.